### PR TITLE
docs: Add graphics and fix links

### PR DESCRIPTION
**Pull request by Seventeen17**

### CODE REVIEW

File: docs/zh_CN/dgs/dataloader.md [hunk 1]
```diff
@@ -1,3 +1,183 @@
-# 从数据源加载数据
+# Loading From Your Data Source
 
-（comming soon...）
+By using kafka queues as the entry for streaming graph updates, the Dynamic-Graph-Service (abbreviated as dgs) can
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:3]: "dgs" --> "DGS"

File: docs/zh_CN/dgs/dataloader.md [hunk 2]
```diff
@@ -1,3 +1,183 @@
...
+decouple the different data sources and sampling workers.
+Users can develop their own data loaders to ingest the source data and push graph updates into the output kafka queues
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:5]: ingest the --> ingest from a customized

File: docs/zh_CN/dgs/dataloader.md [hunk 3]
```diff
@@ -1,3 +1,183 @@
...
+for further consuming.
+
+## Rules to Follow
+
+When processing source data in the dataloader, these rules should be followed:
+- The graph schema from source data must be consistent with dgs.
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:11]: from source data --> of the source data

File: docs/zh_CN/dgs/dataloader.md [hunk 4]
```diff
@@ -1,3 +1,183 @@
...
+- A produced kafka message must be a batch of graph updates, defined as a flatbuffers table [RecordBatchRep](https://github.com/alibaba/graph-learn/blob/master/dynamic_graph_service/fbs/record.fbs).
+- Graph updates are partitioned for sampling workers, all records in one batch must have the same data partition id.
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:13]: for sampling workers --> among sampling workers

File: docs/zh_CN/dgs/dataloader.md [hunk 5]
```diff
@@ -1,3 +1,183 @@
...
+
+## Dataloader SDK
+
+In practice, it is difficult for users to batch records with dgs graph schema, manage partitioning logic and
+produce partitioned batches into kafka queues with themselves.
+Thus, we provide a dataloader sdk (a c++ lib) to help users to do these things.
+
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:20]: The above paragraph:
"We provide a dataloader sdk (a c++ lib) to help users to batch records with following the specified graph schema in DGS, manage partitioning logics and dispatch partitioned records to the kafka queues."

File: docs/zh_CN/dgs/dataloader.md [hunk 6]
```diff
@@ -1,3 +1,183 @@
...
+Build dataloader sdk with cmake:
+```shell
+$ cd dynamic_graph_service/dataloader
+$ mkdir build && cd build
+$ cmake -DCMAKE_INSTALL_PREFIX=$install_prefix ..
+$ make && make install
+```
+
+use it in CMakeLists.txt:
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:32]: use --> Use

File: docs/zh_CN/dgs/dataloader.md [hunk 7]
```diff
@@ -1,3 +1,183 @@
...
+```cmake
+list (APPEND CMAKE_PREFIX_PATH $install_prefix)
+find_package (DataLoader)
+target_link_libraries (program_name
+  PUBLIC DataLoader::dataloader)
+```
+
+### Initialization
+The dataloader must be initialized before the program runs.
+We provide a simple func to help do this, after deploying dgs, specify the dgs host name and
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:42]: "help do this" --> initialize the data loader.

File: docs/zh_CN/dgs/dataloader.md [hunk 8]
```diff
@@ -1,3 +1,183 @@
...
+the sdk will fetch info from dgs and init automatically.
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:43]: init --> perform the initialization

File: docs/zh_CN/dgs/dataloader.md [hunk 9]
```diff
@@ -1,3 +1,183 @@
...
+refer to [file-loader](https://github.com/alibaba/graph-learn/blob/master/dynamic_graph_service/dataloader/apps/file_loader).
+
+## Data Loading Barrier
+
+Sometimes, users may want to deploy their own data-loading cluster with multiple dataloader instances.
+To help users track the data-loading progress of the cluster, we provide a barrier mechanism to check
+the status of data produced from dataloader to dgs service at a synchronized view.
+
+A barrier is a global state and shared between all dataloader instances in cluster.
+Setting a barrier will insert a checking-point into the output data stream (kafka queues),
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:106]: "checking-point " --> "synchronization point"

File: docs/zh_CN/dgs/dataloader.md (hunk 1 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

File: docs/zh_CN/dgs/dataloader.md (hunk 2 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

File: docs/zh_CN/dgs/dataloader.md (hunk 3 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

File: docs/zh_CN/dgs/dataloader.md (hunk 4 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

File: docs/zh_CN/dgs/dataloader.md (hunk 5 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md:20]: done

File: docs/zh_CN/dgs/dataloader.md (hunk 6 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

File: docs/zh_CN/dgs/dataloader.md (hunk 7 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

File: docs/zh_CN/dgs/dataloader.md (hunk 8 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

File: docs/zh_CN/dgs/dataloader.md (hunk 9 above)
Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

File: docs/zh_CN/dgs/dataloader.md [hunk 10]
```diff
@@ -1,3 +1,183 @@
...
+    p.FlushAll();
+}
+```
+
+We provide a default file loader to load a data file according to the data format defined by users,
+refer to [file-loader](https://github.com/alibaba/graph-learn/blob/master/dynamic_graph_service/dataloader/apps/file_loader).
+
+## Data Loading Barrier
+
+Sometimes, users may want to deploy their own data-loading cluster with multiple dataloader instances.
```
Comment (LiSu) [inline docs/zh_CN/dgs/dataloader.md:100]: Delete this sentence?

Comment (Seventeen17) [reply docs/zh_CN/dgs/dataloader.md]: done

**Review by LiSu — APPROVED** [review]

**Review by baoleai — APPROVED** [review]

**Pull request merged**